### PR TITLE
PR to merge "redux-in-react-component" to the "dev"

### PR DESCRIPTION
**Pull request by AliRezaBmeDu**
In this very branch, I have done the following changes.

- Wrap the `App` with the `Provider` component from react-redux
- Pass the configured store into the `Provider`
- Create an `initialState` variable for the `booksSlice` using the given demo list of books.
- Display the books, received from the slice, in a reusable component
- Dispatch actions using `useDispatch`
- Add a `Button` component, which includes:
       - An event handler that adds a book to the books array (with attributes id, title and author)
- Add a `Button` component, which includes:
       - An event handler that removes a book from the books array (by id)

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -11,11 +12,12 @@ const booksSlice = createSlice({
   reducers: {
     addBook: (state, action) => {
       const newBook = action.payload;
-      state.books = state.books.append(newBook);
+      console.log(newBook);
```
Comment (FrederickMih) [inline src/redux/books/booksSlice.js]: - [x] For security purpose, let's kindly always get rid of this link after we are done testing :+1:

File: src/components/Booklist.jsx [hunk 2]
```diff
@@ -1,43 +1,35 @@
-import React, { useState, useEffect } from 'react';
-import PropTypes from 'prop-types';
-import Book from './Book';
+import React from 'react';
+// import PropTypes from 'prop-types';
```
Comment (FrederickMih) [inline src/components/Booklist.jsx]: - [x] Commented codes render your codebase messy. Kindly remove them :+1:

File: src/components/Booklist.jsx [hunk 3]
```diff
@@ -1,43 +1,35 @@
...
+// Booklist.propTypes = {
+//   books: PropTypes.arrayOf(
+//     PropTypes.shape({
+//       id: PropTypes.number.isRequired,
+//       title: PropTypes.string.isRequired,
+//       author: PropTypes.string.isRequired,
+//       category: PropTypes.string.isRequired,
+//     }),
+//   ).isRequired,
+// };
```
Comment (FrederickMih) [inline src/components/Booklist.jsx]: - [x] Kindly check comment above :point_up:

File: src/components/Form.jsx [hunk 4]
```diff
@@ -1,26 +1,30 @@
 import React, { useState } from 'react';
-import PropTypes from 'prop-types';
+// import PropTypes from 'prop-types';
```
Comment (FrederickMih) [inline src/components/Form.jsx]: - [x] Kindly remove all commented codes in your codebase

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @AliRezaBmeDu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- Adding and Removing books is still possible :-1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (AliRezaBmeDu) [reply src/redux/books/booksSlice.js]: Thank You @FrederickMih

File: src/components/Booklist.jsx [hunk 5]
```diff
@@ -1,43 +1,23 @@
...
-    setBookList(books);
-  }, [books]);
-
-  const handleDeleteBook = (bookId) => {
-    // Filter out the book with the given ID
-    const updatedBooks = bookList.filter((book) => book.id !== bookId);
-    setBookList(updatedBooks);
-  };
+const Booklist = () => {
+  // const dispatch = useDispatch();
```
Comment (LionRouge1) [inline src/components/Booklist.jsx]: - [ ] As mentioned by the previous reviewer, please do not push commented out code in your code base.  It makes your project looks messy and unclean. Therefore kindly remove all the commented out code from your code base to make your project looks more professional and clean.

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @AliRezaBmeDu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ibizugbe — APPROVED** [review]
Hi @AliRezaBmeDu 👋🏼👋🏼👋🏼👋🏼, 

approved


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅

- [x] Commented piece of code has been taken away as requested by the previous reviewer👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**